### PR TITLE
SW-4390 Batches by Species -> Add / view batch modal updates

### PR DESCRIPTION
**Pull request by nickgraz**
- Modify 'batch modal' so it can be used with either Species or Nursery as origin
- Move accessions, nurseries, species, and sublocations logic to their own hooks and dropdowns
- Implement the updated modal in both `InventorySeedlingsTable` and `InventorySeedlingsForNurseryTable`

### CODE REVIEW

File: src/components/InventoryV2/view/BatchDetailsModal.tsx
Comment (nickgraz) [inline src/components/InventoryV2/view/BatchDetailsModal.tsx:1]: This was combined with `src/components/InventoryV2/view/BatchDetailsModalForNursery.tsx` and it shows the Species dropdown for Nursery origin and vice versa.

**Comment by constanzauanini** [comment]
Nick I was testing this in Vercel and when adding a nursery filter, and then removing it, I don't get all the results back, it keeps filtered, but maybe it's because latest changes are not merged?

File: src/components/InventoryV2/form/useAccessions.ts
[the remark this answers is not in the document]
Comment (constanzauanini) [reply src/components/InventoryV2/form/useAccessions.ts:1]: For me it's ok here

**Comment by nickgraz** [comment]
Hmm yea I am seeing that in Vercel, but unable to reproduce locally. I am going to push another commit to see if the updated version works as expected

File: src/components/InventoryV2/view/BatchDetailsModal.tsx [hunk 1]
```diff
@@ -92,50 +119,48 @@ export default function BatchDetailsModal(props: BatchDetailsModalProps): JSX.El
...
-      }
+      germinatingQuantity: 0,
+      notReadyQuantity: 0,
+      readyQuantity: 0,
     };
+    if (origin === 'Nursery') {
+      newBatch.facilityId = originNurseryId;
+      newBatch.speciesId = undefined;
+    } else if (origin === 'Species') {
+      newBatch.facilityId = undefined;
```
[the remark this answers is not in the document]
Comment (nickgraz) [reply src/components/InventoryV2/view/BatchDetailsModal.tsx:132]: Right, it is just to make things a bit more obvious, IE, "the species ID must be set in the form somewhere"

**Comment by nickgraz** [comment]
@constanzauanini it seems like the latest push to Vercel fixed the bug we were seeing.

**Pull request merged**